lib/auth/touchid/api.go

```diff
@@ -72,20 +85,46 @@ type CredentialInfo struct {
 	publicKeyRaw []byte
 }
 
+var (
+	cachedDiag   *DiagResult
+	cachedDiagMU sync.Mutex
+)
+
 // IsAvailable returns true if Touch ID is available in the system.
-// Presently, IsAvailable is hidden behind a somewhat cheap check, so it may be
-// prone to false positives (for example, a binary compiled with Touch ID
-// support but not properly signed/notarized).
-// In case of false positives, other Touch IDs should fail gracefully.
+// Typically, a series of checks is performed in an attempt to avoid false
+// positives.
+// See Diag.
 func IsAvailable() bool {
-	// TODO(codingllama): Consider adding more depth to availability checks.
-	//  They are prone to false positives as it stands.
-	return native.IsAvailable()
+	// IsAvailable guards most of the public APIs, so results are cached between
+	// invocations to avoid user-visible delays.
```

Comment:
Let me get you a number.

Another reason to skip multiple calls is that we do make a SecKeyCreateRandomKey as a way of testing, and I'd rather not to a bunch of those needlessly.

Reply:
It takes about 35ms, which seems reasonable given it involves somewhat slow operations. I can't notice the delay, but it could pile up to perceptible numbers without caching (100ms+).

Reply:
That's a good question. I could put an experiment together, but like you said there isn't much we can do regardless. I'll update this thread if I do, but no promises.